File: Problemas/1335A - Candies and Two Sisters/1335A_Candies_And_Two_Sisters.py
# 1335A - Candies and Two Sisters
# https://codeforces.com/problemset/problem/1335/A

# Input:
# 1) pruebas = Número de casos de pruebas
pruebas = int(input())

# Itera sobre cada caso de prueba
for prueba in range(0,pruebas):

    # Input:
    # 2) dulces = Número de dulces
    dulces = int(input())

    # Si el número de dulces es par, entonces hay (N/2) - 1 formas de repartirlos
    if dulces % 2 == 0:
        print(int((dulces / 2) - 1))
    
    # Si el número de dulces es impar, entonces hay (N-1) / 2 formas de repartirlos
    else:  
        print(int((dulces - 1) / 2))

File: Problemas/1335A - Candies and Two Sisters/1335A_Candies_And_Two_Sisters.cpp
// XXXA - 
// https://codeforces.com/problemset/problem/XXX/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) pruebas - Número de casos de prueba
    int pruebas;
    cin >> pruebas;

    for(int prueba_actual = 0; prueba_actual < pruebas; prueba_actual++) {
        
        // Input:
        // 2) dulces - Número de dulces
        int dulces;
        cin >> dulces;

        // Si el número de dulces es par, entonces hay (N/2) - 1 formas de repartirlos
        // Si el número de dulces es impar, entonces hay (N-1) / 2 formas de repartirlos
        if(dulces % 2 == 0) {
            int repartos = (dulces / 2) - 1;
            cout << repartos << endl;
        } else {
            int repartos = (dulces - 1) / 2;
            cout << repartos << endl;
        }
    }
    
    return 0;
}
